### ISSUE TITLE
Send location data when phone's battery is about to die (Android, iOS maybe)

### ISSUE THREAD

**Issue by TCattd** [prey/prey-android-client#28]
As i commented on twitter directly to Tomás https://twitter.com/TCattd/status/256018438403813376

Will be nice to have Prey send a last report, specifically with the current phone's location, when battery charge is low or about to die (maybe with % threshold being an user option).

This idea can be expanded to send a report every time the phone it's powered off by the normal way (hold power button -> power off) before the phone completly powers off... just in case thieve turn it off like normal people do it.

That. Please.

Hope can be implemented in some future ;)

**Comment by jasonspiro** [comment]
Good idea; thank you for suggesting it to the Prey folks.

"Lookout Security & Antivirus" already has your first idea implemented on both Android and iOS.  When a device displays an initial low battery alert (at about battery 15% or so), Lookout stores a location report in the customer's Lookout account.  For more information, please see <https://faq.lookout.com/topics/21487959-what-is-signal-flare>.

For now, you can run both Prey and Lookout at the same time.  If you want, you can disable Lookout's antivirus features, and only enable the lost-device-location features.

But Lookout is closed-source.  The long-term solution is to send Prey a set of patches to implement your idea.  That way people only have to install one anti-theft app, not two.  :)

**Comment by jasonspiro** [comment]
Good idea.

Lookout Mobile Security has offered a [similar feature](https://faq.lookout.com/topics/21487959-what-is-signal-flare) for years.  This is why I've lately recommended to others that they should also install the free version of Lookout, even though it's closed-source.

It's been two and a half years since this issue was filed.  @fanuneza, could you please assign this to _somebody_?  :)

**Comment by lemavri** [comment]
:+1: to signal flare.

> every time the phone it's powered off by the normal way.

I'm no Android developer, but I _think_ this cannot be done. Most likely the OS does not wait for the apps to run something before turning off :(

**Comment by TCattd** [comment]
@lemavri but you can (on android) read the battery level, and do something.

And you don't really need to trigger something when just 1% left or just before the phone turn itself off. 

Prey could just read the battery level, and after a certain threshold is reached (user configurable?), trigger the last update to Prey's servers. Simple (i think).

**Comment by TCattd** [comment]
@lemavri http://developer.android.com/reference/android/content/Intent.html#ACTION_SHUTDOWN ?

**Comment by lemavri** [comment]
Yes, I know there's a broadcast, but as the documentation states: 
>  Once the broadcast is complete, the final shutdown will proceed and all unsaved data lost  

there's not enough time to, for example, send a report or even a notification to the servers.

On the other hand, to do something when the device is at X% of battery is totally loable and we've discussed of implementing it :), so hopefully there might be good news very soon about this.

**Comment by TCattd** [comment]
A battery % threshold trigger it's fine and nice :)
Should work.

Thanks for considering it.

**Comment by jasonspiro** [comment]
@camilaburgos, you closed this issue without posting any message.

Has the feature been implemented?

And, if not, why close the issue?
